src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -289,7 +292,11 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                             if (refreshAll)
                             {
                                 // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
-                                applicationSettings = await LoadAll(client, cancellationToken).ConfigureAwait(false);
+                                // Preserve reference to loaded data with deep copy into serverData before calling mapper functions
+                                Dictionary<string, ConfigurationSetting> data = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
+                                watchedSettings = await LoadKeyValuesRegisteredForRefresh(client, data, cancellationToken).ConfigureAwait(false);
+                                serverData = data.ToDictionary(kvp => kvp.Key, kvp => new ConfigurationSetting(kvp.Value.Key, kvp.Value.Value, kvp.Value.Label, kvp.Value.ETag));
+                                mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);

```

Comment:
`mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);` doesn't need to execute with failover policy, I'd suggest moving it to the if/else after failover policy execution.